packages/theme_tailor/lib/src/generator/theme_tailor_generator.dart

```diff
@@ -103,7 +126,22 @@ class _TailorClassVisitor extends SimpleElementVisitor {
         }
       }
 
-      fields[propName] = Field(propName, coreIterableGenericType(element.type));
+      fields[propName] = Field(
+        name: propName,
+        type: coreType,
+        isAnotherTailorTheme: hasTailorComponentAnnotation,
+        isThemeExtension: isThemeExtension,
+        themeExtensionFields: themeExtensionFields,
```

Comment:
I'd say if there is annotation `@Component` on the field we can assume it conforms to `ThemeExtension<T>` interface therefore can safely call lerp on it. I can see that it can be a bit problematic to determine type of class that is not yet generated (That's I think it's good idea to add annotation of field directly)

```suggestion
        isThemeExtension: isThemeExtension,
```

When it comes to the 'isAnotherTailorTheme' it should not matter 🤔